File: es_12.cpp
// 12. Chiedere in input un numero n positivo,
// inserire n numeri in input e calcolarne la media.

#include <iostream>
using namespace std;

int main() {
    int n_iter, somma = 0;
    float media;

    do {
        cout << "Inserisci un numero positivo: ";
        cin >> n_iter;
    } while (n_iter < 0);

    for (int i = 1; i <= n_iter; i++) {
        int numero;
        cout << "Inserisci un numero: ";
        cin >> numero;

        somma += numero;
    }

    media = float(somma) / n_iter;

    cout << "La somma è: " << somma << endl
         << "La media è: " << media << endl;

    return 0;
}

File: Python/es_12.py
from itertools import count

# 12. Chiedere in input un numero n positivo,
# inserire n numeri in input e calcolarne la media.

while True:
    n_iter = int(
        input('Inserisci un numero n > 0: ')
    )
    
    if n_iter > 0: break
    else: print('Numero non valido')

somma = 0 
for i in count (start=1, step=1):
    n = int(
        input(
            f'({i} di {n_iter}): Inserisci un numero: '
        )
    )
    
    somma += n
    
    
    if i >= n_iter: break

print(
    f'{somma / n_iter}'
)
